AGENTS.md

```diff
@@ -1,21 +1,183 @@
+- Debug: `-gcflags=all="-N -l"`, `-race`, `-tags=debug`
+
+## Lint Commands
+
+```bash
+make lint           # Run golangci-lint (v2 config)
+golangci-lint run   # Direct invocation
+```
+
+The project uses golangci-lint v2 with most linters enabled. Key disabled linters:
+- `cyclop`, `funlen`, `gocognit` - complexity limits
+- `exhaustruct` - struct field exhaustiveness
+- `varnamelen`, `wsl` - style preferences
+
+Formatters enabled: `gofmt`, `gofumpt`, `goimports`
+
+## Test Commands
+
+### Go Tests
+
+```bash
+go test -race ./...                        # Run all tests
+go test -race -run TestName ./package      # Single test
+go test -race -v -run TestFilter ./sel     # Verbose single test
+```
+
+### Python Tests (E2E)
+
+```bash
+poetry run pytest                                      # All tests
+poetry run pytest tests/test_file.py::test_name       # Single test
+poetry run pytest --runslow                            # Include slow tests
+```
+
+Python tests require: `TEST_SOURCE_URI`, `TEST_TARGET_URI`, `TEST_PCSM_URL` env vars.
 
 ## Code Style Guidelines
-- **Imports**: Group stdlib, third-party, local (separated by blank lines). Use `goimports` for formatting.
-- **Error Handling**: Use custom `errors` package (`errors.Wrap`, `errors.New`, `errors.Join`). Always wrap errors with context. Return `nil` for nil errors in Wrap.
-- **Types**: Prefer explicit types. Use generics where appropriate. Document exported types with comments.
-- **Naming**: Use camelCase for unexported, PascalCase for exported. Acronyms stay uppercase (e.g., `URI`, `HTTP`).
-- **Comments**: Document all exported functions/types with godoc format. Start with the function/type name.
-- **Linting**: Follow golangci-lint v2 with most linters enabled (see `.golangci.yml`). Use `//nolint:lintername` sparingly with justification.
-- **Context**: Pass `context.Context` as first parameter. Use `util.CtxWithTimeout` for timeout operations.
-- **Formatting**: Use `gofmt` and `gofumpt`. Max line length flexible but keep readable. Use tabs for indentation.
-- **Testing**: Use `testify` for assertions. Add `-race` flag. Python tests use `pytest` with `pymongo`.
+
+### Imports
+
+Group imports: stdlib, third-party, local (separated by blank lines):
+```go
+import (
+    "context"
+
+    "go.mongodb.org/mongo-driver/v2/bson"
+
+    "github.com/percona/percona-clustersync-mongodb/errors"
+)
+```
+
+### Error Handling
+
+Use the custom `errors` package (not stdlib `errors`):
+```go
+errors.New("message")
+errors.Wrap(err, "context")      // Returns nil if err is nil
+errors.Wrapf(err, "fmt %s", val)
+errors.Join(err1, err2)
+errors.Is(err, target)
+```
+When wrapping is not needed (e.g., returning driver errors directly), use `//nolint:wrapcheck`.
+
+### Naming Conventions
+
+- **Unexported**: `camelCase` (e.g., `sourceCluster`, `nsFilter`)
+- **Exported**: `PascalCase` (e.g., `CloneStatus`, `NewRepl`)
+- **Acronyms**: Stay uppercase (e.g., `URI`, `HTTP`, `PCSM`, `ID`)
```

Comment:
Have you come across issues with naming? Is it needed to be explicit about this?